### ISSUE TITLE
AMD CPU limitation

### ISSUE THREAD

**Issue by mtaylorDW** [MilesCranmer/PySR#158]
**Describe the bug**
AMD CPU produces significantly worse performance than Intel CPUs in the "Simple PYSR Example" from  [pysr_demo.ipynb](https://github.com/MilesCranmer/PySR/blob/master/examples/pysr_demo.ipynb).  AMD appears limited to utilize one core while Intel can utilize all cores. 

**Additional context**
Tested CPU / OS:
- Ryzen 5900x (12 cores / 24 threads): 6% CPU usage (using 1 core, 1x baseline)
- i5 2020 macbook pro (4 cores / 8 threads): 60% CPU usage (using all cores, 2.1 x faster than the AMD)
- i5 6600k windows 10 desktop (4 cores / 4 threads): 80% CPU usage (using all cores,1.2 x faster than the AMD)

 Running the PyTorch trainer In [12], with gpus=0 utilizes all CPU cores on all systems tested.

**Version (please complete the following information):**
- OS: macOS 12.4, intel (OK), Windows 10, intel (OK), Windows 11, AMD (Problem)
- Version: PYSR 0.9.3
- Python 3.9 venv in pycharm 2022.1.3

- Does the bug appear with the latest version of PySR? 
Yes

**Configuration**
- What are your PySR settings?
Exactly the Simple PySR example found in [pysr_demo.ipynb](https://github.com/MilesCranmer/PySR/blob/master/examples/pysr_demo.ipynb)

```python
default_pysr_params = dict(
    populations=30,
    procs=4,
    model_selection="best",
)
# Learn equations
model = PySRRegressor(
    niterations=30,
    binary_operators=["plus", "mult"],
    unary_operators=["cos", "exp", "sin"],
    **default_pysr_params
)

model.fit(X, y)
```

- What dataset are you running on? 
```python
# Dataset
np.random.seed(0)
X = 2 * np.random.randn(100, 5)
y = 2.5382 * np.cos(X[:, 3]) + X[:, 0] ** 2 - 2
```

**Error message**
No message, just performance loss on the AMD chip.

**Comment by MilesCranmer** [comment]
By the way, since there are 8 threads available on your macbook pro, I would set `procs=8`.

(the default setting for `procs` will automatically use the maximum number of cores).

**Comment by mtaylorDW** [comment]
Hi @MilesCranmer , thanks for getting back so soon.  I'm running **julia 1.7.3** on both and it sees **24** threads on the AMD and 8 on the intel MBP.

**Comment by mtaylorDW** [comment]
Thanks @MilesCranmer , `multithreading=False` gave a 2-3x increase in cpu utilization on the AMD machine, utilizing 20% of the CPU resources and doubles the MBP's speed in cycles per second.  I was then able to saturate the CPU by cranking up the `population_size` parameter, which previously had no effect.
